src/View/TemplateAwareViewInterface.php

```diff
@@ -0,0 +1,23 @@
+<?php
+declare(strict_types=1);
+namespace TYPO3Fluid\Fluid\View;
+
+/*
+ * This file belongs to the package "TYPO3 Fluid".
+ * See LICENSE.txt that was shipped with this package.
+ */
+
+/**
+ * Optional addition to ViewInterface if the view deals with template files.
+ *
+ * @api
+ */
+interface TemplateAwareViewInterface
+{
+    /**
+     * @param string A template name to render, e.g. "Main/Index"
+     * @return string The rendered view
+     * @api
+     */
+    public function render(string $templateName = '');
```

Comment:
Hm, I'd rather prefer a `renderTemplate()` or `renderTemplateName()`. This would avoid the implicit override of `ViewInterface::render()` and make for a clearer distinction/API.

Reply:
That wouldn't reflect the current given situation in AbstractTemplateView / TemplateView. Do you really want to change this, and add a second method to TemplateView that has an identical signature?

Reply:
I see what you mean. This by itself is already a violation of the interface IMO even if PHP permits this.

Can you add more context about the issue you are trying to solve? E.g. why type-hinting to `AbstractTemplateView` instead of `ViewInterface` is not an option.

Reply:
Well, in general one should code against interfaces, not implementations. The TYPO3 core type hints ViewInterface at a couple of places, and that's the way to go. It's just that render('Foo') is then not understood.
The patch simply adds a second interface to this, making this argument visible. This is fully PHP compatible and fine with liskov: When an interface declares the method, a second interface can add optional arguments.
I see no reason to not pick this patch.

Reply:
Make sense then, especially if it helps making at least the runtime invocation of `render()` with an argument explicit.